0001-kaa-protocol/README.md

```diff
@@ -142,16 +120,97 @@ Thus, the resource paths for them will start with `kp1/<appversion_name>/<extens
 
 The rest of the resource path is extension-specific and is described in other RFCs defining KP protocol extensions.
 
-## Extension design guidelines
-While extensions have all the freedom to define their own resource hierarchies, payload format, and communication template, they all need a set of rules to make them uniform.
-
-Examples of extension-specific paths:
+Examples of extension-specific Resource Paths:
 ```
 /<endpoint_token>/json
 /protobuf/<scheme_id>
-/json/status
+/json
+```
+
+## Request/response pattern
+Many extensions require request/response style communication, which is natively supported by CoAP, but not MQTT.
+The following section describes request/response communication pattern for MQTT and CoAP.
+
+Note that some requests do not require a full response but only an acknowledgement of reception.
+In that case, KP over MQTT SHOULD use PUBLISH acknowledgement only (PUBACK, PUBREC, PUBREL packets), as defined per MQTT standard.
+
+### Request direction
+For CoAP, server-initiated requests require implementing a server on the client nodes.
+That also implies there must be an open path from server to client, which may require UDP hole punching.
+
+Given that, server-initiated requests are NOT RECOMMENDED.
+
+### Request ID
+CoAP message has a built-in Token which is used to match responses to requests.
+
+MQTT PUBLISH packet includes a Packet Identifier.
+However, Packet Identifiers are not stable; that means, when a PUBLISH packet is re-transmitted by MQTT broker or MQTT gateway, the Packet Identifier may be changed.
+Thus, MQTT Packet Identifiers are not suitable for matching responses to requests.
+
+KP over MQTT MUST NOT use Packet Identifiers to match responses to requests.
+
+KP can not make assumptions about payload format, thus it cannot define the format for passing Request IDs within the message payload.
+Thus, Request IDs are embedded within the topic name.
+
+In case the last topic level in MQTT topic is a positive integer number, it MUST be interpreted as a Request ID.
+
+For example, the request sent to `<endpoint_token>/update/42` topic is a request to `<endpoint_token>/update` Resource Path with Request ID `42`.
+
+If Request ID is not present in the MQTT topic, the response SHOULD NOT be published.
```

Comment:
Does this sentence mean that in case of invalid EP token, app version name or extension instance name and absence of request ID the server SHOULD  NOT publish the error response?